internal/venice-common/src/main/java/com/linkedin/venice/pubsub/PubSubPositionTypeRegistry.java

```diff
@@ -0,0 +1,296 @@
+ * {@link ApacheKafkaOffsetPosition}, rejecting any attempts to override them with mismatched types.
+ * <p>
+ * <b>This registry is immutable after construction</b>. The internal mappings cannot be modified once built,
+ * ensuring thread safety and predictable behavior throughout the lifetime of the instance.
+ * <p>
+ * This class is intended for use in components that need to serialize, deserialize, or resolve
+ * {@link PubSubPosition} types based on their type IDs or class names.
+ */
+public class PubSubPositionTypeRegistry {
+  public static final Logger LOGGER = LogManager.getLogger(PubSubPositionTypeRegistry.class);
+
+  /**
+   * Reserved type IDs for internal standard {@link PubSubPosition} implementations.
+   * <p>
+   * These constants define special type IDs used to identify symbolic or built-in PubSub positions.
+   * They are reserved for internal use and must not be reused, overridden, or assigned to custom
+   * {@link PubSubPosition} implementations by client code.
+   * <p>
+   * Reserved IDs:
+   * <ul>
+   *   <li>{@code -2} —> {@link PubSubSymbolicPosition#EARLIEST}: Marker for the earliest retrievable position in a partition</li>
+   *   <li>{@code -1} —> {@link PubSubSymbolicPosition#LATEST}: Marker for the latest retrievable position in a partition</li>
+   *   <li>{@code  0} —> {@link ApacheKafkaOffsetPosition}: Standard Kafka offset-based position</li>
+   * </ul>
+   * <p>
+   * {@link #POSITION_TYPE_INVALID_MAGIC_VALUE} is used as a sentinel value to indicate that a position
+   * type ID has not yet been assigned or initialized. It must never be used as a valid position type.
+   */
+  public static final int EARLIEST_POSITION_RESERVED_TYPE_ID = -2;
+  public static final int LATEST_POSITION_RESERVED_TYPE_ID = -1;
+  public static final int APACHE_KAFKA_OFFSET_POSITION_TYPE_ID = 0;
+  public static final int POSITION_TYPE_INVALID_MAGIC_VALUE = Integer.MIN_VALUE;
+
+  /**
+   * A predefined map of reserved PubSub position type IDs to their corresponding fully qualified factory class names.
+   * <p>
+   * This map is intended for internal use to bootstrap or validate known position implementations.
+   * It includes all reserved positions that are part of the standard Venice PubSub position model.
+   * <p>
+   */
+  public static final Int2ObjectMap<String> RESERVED_POSITION_TYPE_ID_TO_CLASS_NAME_MAP;
+
+  static {
+    Int2ObjectMap<String> tempMap = new Int2ObjectOpenHashMap<>(3);
+    tempMap.put(EARLIEST_POSITION_RESERVED_TYPE_ID, EarliestPositionFactory.class.getName());
+    tempMap.put(LATEST_POSITION_RESERVED_TYPE_ID, LatestPositionFactory.class.getName());
+    tempMap.put(APACHE_KAFKA_OFFSET_POSITION_TYPE_ID, ApacheKafkaOffsetPositionFactory.class.getName());
+
+    Int2ObjectMap<PubSubPositionFactory> factoryMap = new Int2ObjectOpenHashMap<>(3);
+    factoryMap.put(EARLIEST_POSITION_RESERVED_TYPE_ID, new EarliestPositionFactory(EARLIEST_POSITION_RESERVED_TYPE_ID));
```

Comment:
Just wondering...Why can't we just refer EARLIEST_POSITION_RESERVED_TYPE_ID in EarliestPositionFactory and same for Latest?
This param passing looks unnecessary as we should not have other type ID.

Reply:
got it..sure let's keep it as what it is